R/fit.R

```diff
@@ -312,10 +318,22 @@ mmrm <- function(formula,
       ))
     }
   }
-
-  covbeta_fun <- h_covbeta_fun(fit)
-  fit$jac_list <- h_jac_list(covbeta_fun, fit$theta_est)
-
+  fit$method <- method
+  if (method == "Satterthwaite") {
+    covbeta_fun <- h_covbeta_fun(fit)
+    fit$jac_list <- h_jac_list(covbeta_fun, fit$theta_est)
+  } else {
+    kr_comp <- h_get_kr_comp(fit$tmb_data, fit$theta_est)
+    fit$kr_comp <- kr_comp
```

Comment:
would avoid temporary `kr_comp` here, just do this in one line and use `fit$kr_comp` in below code